hibernate-core/src/test/java/org/hibernate/orm/test/jpa/criteria/CriteriaSelectJoinTest.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ * Copyright Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.orm.test.jpa.criteria;
+
+import java.util.List;
+
+import org.hibernate.testing.orm.junit.EntityManagerFactoryScope;
+import org.hibernate.testing.orm.junit.JiraKey;
+import org.hibernate.testing.orm.junit.Jpa;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.Id;
+import jakarta.persistence.OneToOne;
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.criteria.Join;
+import jakarta.persistence.criteria.JoinType;
+import jakarta.persistence.criteria.Root;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+@Jpa(
+		annotatedClasses = {
+				CriteriaSelectJoinTest.Parent.class,
+				CriteriaSelectJoinTest.Child.class,
+		}
+)
+@JiraKey( "HHH-18628" )
+public class CriteriaSelectJoinTest {
+
+	@BeforeEach
+	public void setUp(EntityManagerFactoryScope scope) {
+		scope.inTransaction(
+				entityManager -> {
+					Child child = new Child( 1l );
+					Parent parent = new Parent( 1l, child );
+					entityManager.persist( parent );
+				}
+		);
+	}
+
+	@AfterEach
+	public void tearDown(EntityManagerFactoryScope scope) {
+		scope.inTransaction(
+				entityManager -> {
+					entityManager.createQuery( "delete from Child" ).executeUpdate();
+					entityManager.createQuery( "delete from Parent" ).executeUpdate();
+				}
+		);
+	}
+
+
+	@Test
+	public void testCriteriaJoin(EntityManagerFactoryScope scope) {
+		scope.inTransaction(
+				entityManager -> {
+					CriteriaBuilder cb = entityManager.getCriteriaBuilder();
+
+					CriteriaQuery<Child> query = cb.createQuery( Child.class );
+					Root<Parent> parent = query.from( Parent.class );
+					Join<Parent, Child> child = parent.join( "child", JoinType.INNER );
+					query.select( child );
+
+					List<Child> children = entityManager.createQuery( query ).getResultList();
+					assertThat( children ).isNotNull();
```

Comment:
Would be nice to also test a `JoinType.LEFT` version of the query, just to ensure we preserve the correct behavior in all cases.

Also, maybe adding a more meaningful assertion on the actual data returned by the query would add value.

Reply:
Thanks @mbladel applied your suggestions.